src/libs/actions/Link.ts

```diff
@@ -64,6 +66,40 @@ function openOldDotLink(url: string) {
     );
 }
 
+function buildTravelDotURL(spotnanaToken?: string, postLoginPath?: string): Promise<string> {
+    return Promise.all([Environment.getTravelDotEnvironmentURL(), Environment.getSpotnanaEnvironmentTMCID()]).then(([environmentURL, tmcID]) => {
+        const authCode = spotnanaToken ? `authCode=${spotnanaToken}` : '';
+        const redirectURL = postLoginPath ? `redirectUrl=${Url.addLeadingForwardSlash(postLoginPath)}` : '';
+        const tmcIDParam = `tmcId=${tmcID}`;
+
+        const paramsArray = [authCode, tmcIDParam, redirectURL];
+        const params = paramsArray.filter(Boolean).join('&');
+        const travelDotDomain = Url.addTrailingForwardSlash(environmentURL);
+        return `${travelDotDomain}auth/code?${params}`;
+    });
+}
+
+/**
+ * @param postLoginPath When provided, we will redirect the user to this path post login on travelDot. eg: 'trips/:tripID'
+ */
+function openTravelDotLink(policyID: OnyxEntry<string>, postLoginPath?: string) {
+    if (policyID === null) {
+        return;
+    }
+
+    const parameters: GenerateSpotnanaTokenParams = {
+        policyID,
+    };
+
+    asyncOpenURL(
+        // eslint-disable-next-line rulesdir/no-api-side-effects-method
+        API.makeRequestWithSideEffects(SIDE_EFFECT_REQUEST_COMMANDS.GENERATE_SPOTNANA_TOKEN, parameters, {})
+            .then((response) => (response?.spotnanaToken ? buildTravelDotURL(response.spotnanaToken, postLoginPath) : buildTravelDotURL()))
+            .catch(() => buildTravelDotURL()),
```

Comment:
I think that we need to throw some sort of error here (and line 97) if we don't get the token. Otherwise it will just error out on Spotnana's side right?

Reply:
@stitesExpensify yes that's a good point 👍 
right now, it'll show "Something went wrong" error on spotnana side.

I'm not sure where would be a good place to show the error. This method can be called from any component.

Reply:
If we throw an error here, we'll have to add try catch everywhere this navigation function is called

Reply:
I wasn't thinking an actual throw, but more like a red brick road notification. Maybe for now it's okay to just let Spotnana show an error? I'm not sure that the user would know what to do next though.. cc @twisterdotcom

Reply:
I think it's fine to leave it as the Spotnana error for now. Can we log it at least?

Reply:
I think a small pre-design would be nice. I created an issue to handle errors - https://github.com/Expensify/App/issues/43780